ddl/ddl_api.go

```diff
@@ -205,9 +208,47 @@ func (d *ddl) ModifySchemaDefaultPlacement(ctx sessionctx.Context, stmt *ast.Alt
 	return errors.Trace(err)
 }
 
-func (d *ddl) ModifySchemaSetTiFlashReplica(ctx sessionctx.Context, stmt *ast.AlterDatabaseStmt, tiflashReplica *ast.TiFlashReplicaSpec) error {
+func getBatchPendingTiFlashCount(ctx sessionctx.Context) uint32 {
+	sessionVars := ctx.GetSessionVars()
+	count, err := variable.GetSessionOrGlobalSystemVar(sessionVars, variable.TiDBBatchPendingTiFlashCount)
+	if err != nil {
+		logutil.BgLogger().Error("can not get TiDBBatchPendingTiFlashCount", zap.Error(err))
+		return variable.DefTiDBBatchPendingTiFlashCount
+	}
+	c, err := strconv.ParseUint(count, 10, 64)
+	if err != nil {
+		logutil.BgLogger().Error("can not parse TiDBBatchPendingTiFlashCount", zap.Error(err))
+		return variable.DefTiDBBatchPendingTiFlashCount
+	}
+	return uint32(c)
+}
+
+// getPendingTiFlashTableCount counts unavailable TiFlash replica by iterating all tables in infoCache.
+func (d *ddl) getPendingTiFlashTableCount(originVersion *int64, pendingCount *uint32) {
+	is := d.infoCache.GetLatest()
```

Comment:
Why not `GetInfoSchemaWithInterceptor`, does it break tests?

Reply:
It comments `Please don't use this function, it is used by TestParallelDDLBeforeRunDDLJob ...` now. I can use it.

Reply:
That is not true for now. It is introduced by #8890, an old PR. You could just delete the old comments and use it.

Reply:
fixed